### ISSUE TITLE
Default behavior for clicking links

### ISSUE THREAD

**Issue by quuu** [silverbulletmd/silverbullet#332]
Hi,

Would love to hear your thoughts on this suggestion: silverbullet feels great to use, but one thing is when editing external links/ page links, it's sometimes difficult to have to click past the link, and then arrow key back into it to edit as clicking on the link itself will open up that external link/ page.

I'm wondering what you think of removing the onclick for opening external links/ page links, but allow opening the link only if Command (on Mac), or Control (on Windows) is held down.

**Comment by zefhemel** [comment]
If you hold Alt and click on a link it moves the cursor there. Does that work too?

**Comment by quuu** [comment]
> If you hold Alt and click on a link it moves the cursor there. Does that work too?

yea that worked, but was taking inspiration from how VS Code does it -- as editing text being the first priority, clicking through text doesn't disrupt the editing experience, only when you're intentionally holding down Cmd and clicking does it change the behavior of text navigation

<img width="231" alt="CleanShot 2023-01-23 at 21 13 15@2x" src="https://user-images.githubusercontent.com/32676955/214217007-ab9a08c4-1021-43ed-9079-eb151bc00db3.png">

**Comment by zefhemel** [comment]
Yeah, I don’t think there’s a right answer here. Earlier versions of SB did what you suggest: you had to Cmd/Ctrl-click links to navigate there. However since I think that in SB we more often want to follow links than edit them, I decided to default to following links on click, and having Alt-click as a navigation option. I think this works especially well when you’re in “browsing/reading” mode. And on silverbullet.md where SB itself is used to render the content, it would just be strange not to have this behavior, because it’s not “webby.” 

I think this is different in VS Code. You likely want to usually move your cursor somewhere more than you navigate (or want ot jump to definition). Code editors have always behaved this way.

**Comment by danberindei** [comment]
How about changing the behavior depending on whether a paragraph is being edited or not, so I can click anywhere in the paragraph/directive or in the directive to get the edit mode (pending https://github.com/silverbulletmd/silverbullet/issues/424), then click in the middle of the link to edit the link?

Of course, it wouldn't make sense to have two different modifiers to do essentially the same thing, so I'd suggest replacing Alt with Ctrl/Cmd to move the cursor without navigating outside the edited paragraph.

**Comment by zefhemel** [comment]
That's doable, but wouldn't it be highly confusing from a UX perspective? Sometimes clicking does one thing and sometimes another depending on where your cursor was before... I don't know...

**Comment by simone-viozzi** [comment]
> Earlier versions of SB did what you suggest: you had to Cmd/Ctrl-click links to navigate there.

Is there a setting to revert the old behavior?

**Comment by danberindei** [comment]
Personally I've gotten used to the current behaviour, but it's perhaps easier because I'm not editing Markdown (or other "text with links" formats) in other editors very often. And having an object that looks like a link behave like a link helps with consistency.

There is still one minor annoyance: when I Alt-click on a link, the cursor is placed at the start of the link, not on the character that I clicked on. Similarly, when I press the End key and the line ends with a link, the cursor ends at the beginning of the link, not at the end. 

I sometimes edit a paragraph with multiple links and I think it would be nice to have an "edit mode" command that expands everything in a paragraph/page and keeps everything expanded until a page switch, but I'm not sure how much I'd actually use it.

PS: Thanks @zefhemel for the (partial) fix for #592, I had a similar issue with clicking on the empty space after a link and it seems to be fixed in 0.5.11 (in previous versions it looked like the mouse down event placed the cursor at the end of the link, the link expanded so it was now under the mouse cursor, and then the mouse up event opened the link).

**Comment by gray-space** [comment]
Just to chime in, I'm pulling my heair out with the current behavior. It seemed to me that arrowing into a link triggers an edit mode (you see the markdown link text and the URL) where you are *editing* rather than being in a browsing mode where you follow a link.

The text of the URL is especially bad because it gives no user affordance that clicking in it will open the link. In fact, it gives every indication it's simply editable text: the text has no underline or highlight to indicate it will act as a link and the mouse cursor shows an I-beam instead of a pointing finder that implies a link. 

The Alt key option isn't easily discoverable. In fact, until I saw this thread I didnt realize it worked.

**Comment by dessalines** [comment]
Would it be possible to make this a setting? Often on mobile, its too easy to accidentally click a link, especially if you're trying to select text for copying.

Personally I really like the follow link popup [suggested above](https://github.com/silverbulletmd/silverbullet/issues/332#issuecomment-1401407894) as a sane default, but an option to not follow links would also work.
